python/lsst/ts/salobj/topics/read_topic.py

```diff
@@ -529,7 +541,13 @@ def get_oldest(self):
         return None
 
     async def next(self, *, flush, timeout=None):
-        """Wait for a message, possibly returning the oldest queued message.
+        """Wait for a message, or immediately return the oldest queued message.
+        This is a pop operation and the message will be permanently removed
+        from the queue.
+
+        To ensure the next message is received, the topic should be flushed
+        prior to being populated using the `flush` method, then this method
+        should be called/awaited with the flush attribute set to False
 
         Parameters
         ----------
```

Comment:
I don't understand the "if any" case below. You're saying that regardless of how the `flush` parameter is set the routine will wait for a new message before returning? Even if it returns the oldest message (which would have already been in the queue?

Reply:
The new brief string for `next` is  "Pop and return the oldest message from the queue, waiting for data if the queue is empty." on https://github.com/lsst-ts/ts_salobj/pull/145/files.